commit 979c283a6ecb1ba718790a147b3e910a106cb473
Author: Justin Blanchard <justinb04@aim.com>
Date:   Tue Jan 30 20:13:52 2007 -0500

    update from Justin Blanchard - part 2
    
    * Fix a bug that prevents active-pen users from right-clicking while hovering.
    * Bug was introduced with passive pen support; fix is simple and has been
    recommended in forums since Oct 2006
    (http://ubuntuforums.org/showpost.php?p=1620646&postcount=15).
    For some reason I don't see a bug report for the issue.

diff --git a/src/xf86Fpit.c b/src/xf86Fpit.c
index d0b9fe3..636d6d0 100644
--- a/src/xf86Fpit.c
+++ b/src/xf86Fpit.c
@@ -262,11 +262,9 @@ static void xf86FpitReadInput(LocalDevic
 			if (buttons & SW2) buttons=0; /* the pen was lifted, so no buttons are pressed */
 			else if (buttons & SW3) buttons=SW3; /* the "right mouse" button was pressed, so send down event */
 			else if (prox) buttons=SW1; /* the "left mouse" button was pressed and we are not hovering, so send down event */
-			else buttons=0; /* We are in hover mode, so no buttons */
-		}
-		else { /* the active pen's buttons map directly to the mouse buttons */
-			if (!prox) buttons=0; /* We are in hover mode, so no buttons */
+			else buttons=0; /* We are in hover mode, so not left-clicking. */
 		}
+		/* the active pen's buttons map directly to the mouse buttons. Right-click may happen even in hover mode. */
 	
 		/* DBG(2, ErrorF("%02d/%02d Prox=%d SW:%x Buttons:%x->%x (%d, %d)\n",
 			loop,priv->fpitIndex,prox,priv->fpitData[loop]&BUTTON_BITS,priv->fpitOldButtons,buttons,x,y));*/
